File: bench/reals.lua
-- The Lua twin of reals.sl. Lua's float subtype is a double, the same as slate's real, and `i * 1.5`
-- promotes the integer exactly as slate does.

local function run()
    local total, i = 0.0, 0

    while i < 10000000 do
        total = total + i * 1.5 - 0.5
        i = i + 1
    end

    return total
end

print(string.format("%.1f", run()))

File: bench/reals.php
<?php
// The PHP twin of reals.sl. A PHP float is a double and `$i * 1.5` promotes the int, so the
// arithmetic is the same sequence of operations on the same values. `echo` of a float prints
// fourteen significant digits in exponent form at this size, so the answer is formatted explicitly.

function run()
{
    $total = 0.0;
    $i = 0;

    while ($i < 10000000) {
        $total = $total + $i * 1.5 - 0.5;
        $i = $i + 1;
    }

    return $total;
}

printf("%.1f\n", run());
